wordpress-mobile/WordPress-iOS
# Add to Editor tests the ability to close the editor screen when Screen Object fails to load

One identified UI Tests issue happens when the Editor screen takes longer to load. Some elements would load right away like the ones In Navigation Bar, but others may take some more seconds to load, like the editor blocks and the `add-block-button`. 

![image](https://user-images.githubusercontent.com/42008628/152846394-77dc7541-a002-4fd4-b5a2-bf0f44a8be5e.png)

This issue doesn't reproduce consistently so it would most likely pass in the automatic retry, however the `tearDown()` and `setUp` can't handle the Editor screen, so all the following tests fail. 
The purpose of this PR is to enable Edior tests' `tearDown()` to properly close the Editor when the Editor Screen Object fails initializing.

* Changes in `tearDown()`
When the issue happens `editorScreen()` evaluates to `nil`, a check was added and `closeEditorDiscardChanges()` will run in that case .

* Changes in `BlockEditorScreen`
  * `expectedElementGetters`
  The order of the elements was changed so the element that loads first is the first to be checked.
  * Increased timeout 
  The initial idea was not to increase the timeout for time sake, however from the evidences we have, the button did load right before the 7 secs timeout but it was not enough time to check for the `Close` button that was already there right from the start. Also adding 3 more seconds to the execution, in the worst scenario, is still much better than rerunning the test in the automatic retry.

To test:
Running locally:
In BlockEditorScreen, change the add block button name from `add-block-button` to something else.

Running on CI:
I don't know how to reproduce it in CI but I have re-enabled the Editor tests for testing this PR. If you we are lucky the issue will happen and we will still see all the other tests running normally, including the failed test retries.

**P.S.** I'll disable the Editor tests before merging this PR.

## Regression Notes
1. Potential unintended areas of impact
Editor UI Tests

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Ran the tests in CI

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:
